### PR TITLE
fix bounded arrays

### PR DESCRIPTION
**Pull request by dirk-thomas**
At least compiling the generated code for unbounded array passes with this.

Connect to ros2/system_tests#123.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please review.

File: rosidl_generator_py/resource/_msg_support.c.em [hunk 1]
```diff
@@ -84,7 +84,11 @@ nested_type = '%s__%s__%s' % (field.type.pkg_name, 'msg', field.type.type)
   @(nested_type) * dest@(field.name) = ros_message->@(field.name).data;
 @[      else]@
   size_t size@(field.name) = @(field.type.array_size);
+@[        if field.type.is_upper_bound]@
+  @(nested_type) * dest@(field.name) = ros_message->@(field.name).data;
+@[        else]@
   @(nested_type) * dest@(field.name) = ros_message->@(field.name);
+@[        end if]@
```
Comment (mikaelarguedas) [inline rosidl_generator_py/resource/_msg_support.c.em]: I think that you want to call the `__Array__init()` methods for bounded arrays.
I would suggest checking `is_upper_bound` L79 and if the array is bounded, use `min(PySequence_Size(),upper_bound)` to set the array size.

Same below

Comment (dirk-thomas) [inline rosidl_generator_py/resource/_msg_support.c.em]: This PR only fixes the compilation. Getting the code to actually work will take another iteration.

Comment (mikaelarguedas) [inline rosidl_generator_py/resource/_msg_support.c.em]: Alright, in this case let's merge the first iteration and get things fixed before actually merging ros2/system_tests#124

**Comment by mikaelarguedas** [comment]
lgtm

**Pull request merged**